frontend/packages/employee-mobile-frontend/src/components/attendances/GroupSelector.tsx

```diff
@@ -0,0 +1,102 @@
+// SPDX-FileCopyrightText: 2017-2020 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+import React, { Fragment, useContext } from 'react'
+import styled from 'styled-components'
+
+import { SelectionChip } from '@evaka/lib-components/src/atoms/Chip'
+import { Gap } from '@evaka/lib-components/src/white-space'
+import { Success } from '@evaka/lib-common/src/api'
+
+import { AttendanceUIContext } from '~state/attendance-ui'
+import { useTranslation } from '~state/i18n'
+import { AttendanceResponse, Group } from '~api/attendances'
+
+interface GroupSelectorProps {
+  groupIdOrAll: string | 'all'
+  selectedGroup: Group | undefined
+  changeGroup: (groupOrAll: Group | 'all') => void
+}
+
+export default function GroupSelector({
+  groupIdOrAll,
+  selectedGroup,
+  changeGroup
+}: GroupSelectorProps) {
+  const { i18n } = useTranslation()
+
+  const { attendanceResponse } = useContext(AttendanceUIContext)
+
+  function numberOfChildrenPresent(
+    gId: string,
+    res: Success<AttendanceResponse>
+  ) {
+    return res.value.children
+      .filter((child) => child.groupId === gId)
+      .filter((child) => {
+        // return child.status === currentPage?.toUpperCase()
+        return child.status === 'PRESENT'
+      }).length
+  }
+
+  function getTotalChildren(groupId: string) {
+    if (attendanceResponse.isSuccess) {
+      return attendanceResponse.value.children.filter((ac) => {
+        return ac.groupId === groupId
+      }).length
+    }
+    return 0
+  }
+
+  return (
+    <GroupChipWrapper>
+      {attendanceResponse.isSuccess && (
+        <>
+          <SelectionChip
+            text={`${i18n.common.all} (${
+              attendanceResponse.value.children.filter((child) => {
+                return child.status === 'PRESENT'
+              }).length
+            }/${attendanceResponse.value.children.length})`}
+            selected={groupIdOrAll === 'all' ? true : false}
+            onChange={() => changeGroup('all')}
+          />
+          <Gap horizontal size={'xxs'} />
+          {attendanceResponse.value.unit.groups.map((group) => (
+            <Fragment key={group.id}>
+              <SelectionChip
+                text={`${group.name} (${numberOfChildrenPresent(
+                  group.id,
+                  attendanceResponse
+                )}/${getTotalChildren(group.id)})`}
+                selected={
+                  selectedGroup
+                    ? selectedGroup.id === group.id
+                    : groupIdOrAll === 'all'
+                    ? true
```

Comment:
Showing all chips as selected when all-option is selected seems wrong to me, since i feel that then clicking one would deselect that one and leave the others selected. This should behave more like a dropdown or radio button group where only one option can be selected at a time (one of the options just means "all").

Reply:
Yeah, totally forgot about that. Also added a new `ChoiceChip` component that is just `SelectionChip` without the checkmark.